### ISSUE TITLE
[FEAT]: Support for codespaces and dependabot secrets

### ISSUE THREAD

**Issue by kamarcum** [octokit/octokit.rb#1594]
### Describe the need

I'm working with repo secrets and would like to use octokit with all 3 flavors of repo secrets: actions, codespaces, and dependabot.

The APIs are very nearly identical:
- https://docs.github.com/en/rest/codespaces/repository-secrets?apiVersion=2022-11-28
- https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28
- https://docs.github.com/en/rest/dependabot/secrets?apiVersion=2022-11-28

I started a naive implementation of copying [action_secrets](https://github.com/octokit/octokit.rb/blob/main/lib/octokit/client/actions_secrets.rb) and updating request paths. But that would leave us with name collisions when everything is included in the Client.

I'm happy to do the work, but I'm curious what approach the maintainers would like for avoiding name collisions.
Some options:
1. `client.create_or_update_actions_secret(repo, name, options)`
1. `client.create_or_update_secret(repo, name, :codespaces, options)`
1. `client.create_or_update_secret(repo, name, options)` with a special "secret_type" option that would be deleted before being passed on to the request.
1. Something more clever?

Let me know what you think. My preference is option 3 as it's probably the least disruptive.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nickfloyd** [comment]
Hey @kamarcum thanks for all that you're doing to call out the need for consistency and patterns.

TL;DR - `client.create_or_update_actions_secret(repo, name, options)` | Much like what we did [here](https://github.com/octokit/octokit.rb/pull/1588) - reactions are the quintessential example for name collision

Even though it makes a bit more code / work I personally favor esoteric method names over parameter overloading. i.e. Clarity over obscurity.

My reasoning is that I believe that discoverability with APIs and SDK should be paramount (acknowledging that SDKs should also not 1:1 maps for APIs - there's a balance)

SDKs are meant to make consumption of various resources more approachable and also to abstract mundane and the known (like authentication and paging) away so that users don't have to repeat themselves.

All this to say, I feel that we should favor concrete over clever whenever we can - to best serve our new and existing consumers.

Thanks again for this! It's such a good call out! ❤️
